### DISCUSSION TITLE
How to change port on custom plugin lib

### DISCUSSION THREAD

**Discussion by elmalakomar** [mia-platform/community#168] [Q&A] [answered]
Hi,
by default the custom plugin lib launches the service on port 3000.
Is there a way to change the port when running the service in local? 

thanks

**Comment by fredmaggiowski** [comment, marked as answer]
If the service is created from one of Mia-Platform templates, you can change the `HTTP_PORT` environment variable to change the value of the exposed port.

The exposed port is controlled by the `--port` flag of [lc39](https://github.com/mia-platform/lc39/blob/master/docs/cli-flags.md) which is generally controlled by the `HTTP_PORT` environment variabile.

You can checkout this example [Dockerfile](https://github.com/mia-platform-marketplace/Node.js-Custom-Plugin-Template/blob/master/Dockerfile#L38) which configures the flag.
